app/ante/ante.go

```diff
@@ -21,24 +21,46 @@ func NewAnteHandler(
 	channelKeeper *ibckeeper.Keeper,
 ) sdk.AnteHandler {
 	return sdk.ChainAnteDecorators(
-		ante.NewSetUpContextDecorator(), // outermost AnteDecorator. SetUpContext must be called first
-		// reject all tx extensions
+		// Set up the context with a gas meter.
+		// Contract: must be called first.
+		ante.NewSetUpContextDecorator(),
+		// Ensure the tx does not contain any extension options.
 		ante.NewExtensionOptionsDecorator(nil),
+		// Ensure the tx passes ValidateBasic.
 		ante.NewValidateBasicDecorator(),
+		// Ensure the tx has not reached a height timeout.
 		ante.NewTxTimeoutHeightDecorator(),
+		// Ensure the tx memo <= max memo characters.
 		ante.NewValidateMemoDecorator(accountKeeper),
+		// Ensure the tx's gas limit is > the gas consumed based on the tx size.
+		// Side effect: consumes gas from the gas meter.
 		ante.NewConsumeGasForTxSizeDecorator(accountKeeper),
-		// check that the fee matches the gas and the local minimum gas price
-		// of the validator
+		// Ensure the feepayer (fee granter or first signer) has enough funds to pay for the tx.
+		// Side effect: deducts fees from the fee payer. Sets the tx priority in context.
 		ante.NewDeductFeeDecorator(accountKeeper, bankKeeper, feegrantKeeper, checkTxFeeWithValidatorMinGasPrices),
-		ante.NewSetPubKeyDecorator(accountKeeper), // SetPubKeyDecorator must be called before all signature verification decorators
+		// Set public keys in the context for fee-payer and all signers.
+		// Contract: must be called before all signature verification decorators.
+		ante.NewSetPubKeyDecorator(accountKeeper),
+		// Ensure that the tx's count of signatures is <= the tx signature limit.
 		ante.NewValidateSigCountDecorator(accountKeeper),
+		// Ensure that the tx's gas limit is > the gas consumed based on signature verification.
+		// Side effect: consumes gas from the gas meter.
 		ante.NewSigGasConsumeDecorator(accountKeeper, sigGasConsumer),
+		// Ensure that the tx's signatures are valid.
+		// Note: does not consume gas from the gas meter.
 		ante.NewSigVerificationDecorator(accountKeeper, signModeHandler),
```

Comment:
This also checks the nonces (sequence number)

Reply:
Good call! Described in da09fa2766886e8c7e514b7ed95529771a62861f